backend/zoho/zoho.go

```diff
@@ -710,21 +767,40 @@ func (f *Fs) upload(ctx context.Context, name string, parent string, size int64,
 //
 // The new object may have been created if an error is returned
 func (f *Fs) Put(ctx context.Context, in io.Reader, src fs.ObjectInfo, options ...fs.OpenOption) (fs.Object, error) {
-	size := src.Size()
-	remote := src.Remote()
+	existingObj, err := f.NewObject(ctx, src.Remote())
+	switch err {
+	case nil:
+		return existingObj, existingObj.Update(ctx, in, src, options...)
+	case fs.ErrorObjectNotFound:
+		size := src.Size()
+		remote := src.Remote()
+
+		// Create the directory for the object if it doesn't exist
+		leaf, directoryID, err := f.dirCache.FindPath(ctx, remote, true)
+		if err != nil {
+			return nil, err
+		}
 
-	// Create the directory for the object if it doesn't exist
-	leaf, directoryID, err := f.dirCache.FindPath(ctx, remote, true)
-	if err != nil {
-		return nil, err
-	}
+		// use normal upload API for small sizes (<10MiB)
+		if size < 10*1024*1024 {
```

Comment:
On other backends which do this we allow this to be configured with an upload cutoff

Eg s3

https://github.com/rclone/rclone/blob/2d1c2b1f76f44b4db5369bacdd4e5f76feecc128/backend/s3/s3.go#L2119-L2126

```
      --azureblob-upload-cutoff string       Cutoff for switching to chunked upload (<= 256 MiB) (deprecated)
      --b2-upload-cutoff SizeSuffix          Cutoff for switching to chunked upload (default 200Mi)
      --box-upload-cutoff SizeSuffix         Cutoff for switching to multipart upload (>= 50 MiB) (default 50Mi)
      --drive-upload-cutoff SizeSuffix       Cutoff for switching to chunked upload (default 8Mi)
      --hidrive-upload-cutoff SizeSuffix     Cutoff/Threshold for chunked uploads (default 96Mi)
      --oos-upload-cutoff SizeSuffix         Cutoff for switching to chunked upload (default 200Mi)
      --qingstor-upload-cutoff SizeSuffix    Cutoff for switching to chunked upload (default 200Mi)
      --s3-upload-cutoff SizeSuffix          Cutoff for switching to chunked upload (default 200Mi)
      --sharefile-upload-cutoff SizeSuffix   Cutoff for switching to multipart upload (default 128Mi)
```

It would probably be a good idea to add that here just in case users need to tweak it